backends/p4tools/modules/testgen/test/gtest_utils.h

```diff
@@ -47,7 +52,22 @@ class P4ToolsTestCase {
     static void ensureInit();
 };
 
-class P4TestgenTest : public testing::Test {};
+class P4TestgenTest : public testing::Test {
+ public:
+    [[nodiscard]] static std::optional<std::unique_ptr<AutoCompileContext>> SetUp(
```

Comment:
Any reason to return optional of pointer? Why would `nullptr` not work to indicate no context? Presumably `optional<AutoCompileContext>` is out of question because it is immovable, right?

Reply:
I was feeling a little iffy about returning a copyable AutoCompileContext. I ran into lots of compile context mismatches implementing this PR, it is easy to lose track of when one is pushed or popped. 

Maybe we should make the available operations explicit there. 

Iirc, there is no particular reason why I use `std::optional` here except that I prefer to use it over nullptr checks nowadays. I changed it.